cores/arduino/zephyrSerial.cpp

```diff
@@ -20,6 +24,32 @@ size_t arduino::ZephyrSerial::print(char ch){
     return print_char(ch, false);
 }
 
+size_t arduino::ZephyrSerial::print(const int val) {
+    printk("%d",val);
+    return sizeof(int);
+}
+
+size_t arduino::ZephyrSerial::print(double d) {
+    printk("%0.4f",d);
```

Comment:
```suggestion
    printk("%0.2f",d);
```

I think [according to the Arduino reference](https://www.arduino.cc/reference/en/language/functions/communication/serial/print/) the default floating point precision is 2 decimal places.

Also, we should have the same function but for `float` type parameters.

Reply:
You may consider implementing a `size_t arduino::ZephyrSerial::print(double d, const int precision )` precision function and calling that from this function with `2` as the precision. You can do an `snprintk` with the precision as a variable in order to create the format string passed to the `printk` statement:

```
snprintk(buffer, sizeof(buffer), "\%0.%df", precision);
printk(buffer, d);
```